### PR TITLE
Global Styles Sidebar: tweak separator margin

### PR DESCRIPTION
**Pull request by ciampo**
## What?

Tweak the styles of the separator used in the root screen of the Global Styles Sidebar, removing its vertical margin and enforcing its expected color.

## Why?

This is part of a series of tweaks aimed at polishing the Global Styles Sidebar — see #38934 for more details.

## How?

The changes in this PR mainly override [the margin styles](https://github.com/WordPress/gutenberg/blob/c1ea8d60db9f05ebbfbaced50a88bfe30a7a0270/packages/interface/src/components/complementary-area/style.scss#L53-L57) that the `ComplementaryArea` component applies to all children `hr` . In particular, the margin has been changed to be `0`.

The idea is that any vertical spacing around the separator may be sufficiently provided by the padding of the sibling components.

## Testing Instructions

- Open the site editor
- Open global styles sidebar (if not already opened)
- Make sure that the vertical spacing around the separator looks good

## Screenshots or screencast 

| Before | After |
|---|---|
| Screenshot 2022-04-21 at 20 11 09 | Screenshot 2022-04-21 at 20 03 42 |

### CODE REVIEW

File: packages/edit-site/src/components/sidebar/style.scss [hunk 1]
```diff
@@ -136,3 +136,10 @@ $split-border-control-offset: 55px;
 		}
 	}
 }
+
+// Override the `hr` styles defined for the complementary area component
```
Comment (jasmussen) [inline packages/edit-site/src/components/sidebar/style.scss]: Short and sweet. Is this meant to be a temporary override to a point where the HR can be customized on the component level? If yes, then we might want to add a comment to that effect.

File: packages/edit-site/src/components/sidebar/style.scss [hunk 2]
```diff
@@ -136,3 +136,10 @@ $split-border-control-offset: 55px;
...
+// from the `@wordpress/interface` package.
+.edit-site-global-styles-sidebar hr {
+	margin: 0;
+	border-color: rgba(0, 0, 0, 0.1);
```
Comment (jasmussen) [inline packages/edit-site/src/components/sidebar/style.scss]: Why do we need to override the if it's mainly the margin that need zeroing out?

While diving in here, I also noticed that the previous color is wrong. It outputs #f0f0f0 (which maps to $gray-100), but it should have been #e0e0e0 ($gray-200) like the other panel borders.

Depending on your thoughts here (maybe there's a reason you used opacity), I could imagine two paths forward:

1. We remove the border-color override here, and accept the wrong color coming from the component directly, and then update the component to use the correct color (#e0e0e0) in a separate PR.
2. We remove he border-color override here, _and_ fix the component in the same PR. 

What do you think?

**Review by jasmussen — APPROVED** [review]
I left some feedback to address, but still giving approval to a nice small fixup! Thank you!

Before:
before

After:
after

File: packages/edit-site/src/components/sidebar/style.scss (hunk 2 above)
Comment (mirka) [inline packages/edit-site/src/components/sidebar/style.scss]: Hoping we can replace this color value with a CSS var in the near future 🤞 

(Leaving a breadcrumb here for future cleanup #40392)

**Review by mirka — APPROVED** [review]

File: packages/edit-site/src/components/sidebar/style.scss (hunk 2 above)
Comment (jasmussen) [reply packages/edit-site/src/components/sidebar/style.scss]: We use `$gray-200` for similar borders, but I sense there may be a different effort at work here?

File: packages/edit-site/src/components/sidebar/style.scss (hunk 1 above)
Comment (ciampo) [reply packages/edit-site/src/components/sidebar/style.scss]: Not really — the Global Styles Sidebar is already using `CardDivider` for this separator. The issue is that the sidebar internally uses a component from the `@wordpress/interface` package called `ComplementaryArea`, which introduces some style overrides, including [this one on all its children `hr`](https://github.com/WordPress/gutenberg/blob/c1ea8d60db9f05ebbfbaced50a88bfe30a7a0270/packages/interface/src/components/complementary-area/style.scss#L53-L57).

A few solutions that come to mind:

- Artificially increase the specificity of the `CardDivider` styles, so that they would not get overridden by the styles of `ComplementaryArea` — although I wouldn't recommend going down this route, it's a hacky approach that can cause a "race for specificity" in other parts of the repo;
- I guess the better solution would be to somehow rework how separators are used / styled in `ComplementaryArea`, although it may be a difficult task given how widespread the usage of `ComplementaryArea` may be.

Basically, the problem is that separators under `ComplementaryArea` are not encapsulated components, but are instead styles with some "global" css.

File: packages/edit-site/src/components/sidebar/style.scss (hunk 2 above)
Comment (ciampo) [reply packages/edit-site/src/components/sidebar/style.scss]: > maybe there's a reason you used opacity

The separator in the Global Styles sidebar is rendered as a `CardDivider`, which uses `rgba(0, 0, 0, 0.1)` (as [per the config](https://github.com/WordPress/gutenberg/blob/e4252e430a04e0f528142a5a1f56e2b1971ecfce/packages/components/src/utils/config-values.js#L43)) as the color of its border.

The style overrides introduced by `ComplementaryArea` [changed the separator color to `$gray-100`](https://github.com/WordPress/gutenberg/blob/c1ea8d60db9f05ebbfbaced50a88bfe30a7a0270/packages/interface/src/components/complementary-area/style.scss#L55), and therefore I introduced this rule to bring the separator color back to how the original component was intended to be.

> I could imagine two paths forward:

**I will go ahead and drop this color override from this PR.** As for the next steps, there will be a few separate fronts to work on:

1. **Review separator and border colors of `@wordpress/components`**.  We can definitely look at reviewing and updating the color used by components in `@wordpress/components` (e.g. [card](https://github.com/WordPress/gutenberg/blob/e4252e430a04e0f528142a5a1f56e2b1971ecfce/packages/components/src/utils/config-values.js#L43), [surface](https://github.com/WordPress/gutenberg/blob/e4252e430a04e0f528142a5a1f56e2b1971ecfce/packages/components/src/utils/config-values.js#L76), [datepicker](https://github.com/WordPress/gutenberg/blob/c3ff86d88f35ceb92bfc2ad58a335fc9de7e9e09/packages/components/src/date-time/datepicker.scss#L513)...). Although I wasn't involved in the conversation at the time, if I had to guess  that color is currently using transparency to make it blend better in case the background color is not white

2. **Review the `rgba(0, 0, 0, 0.1)` color across Gutenberg.** Depending on the results of the previous step, we should also review the remaining usage of `rgba(0, 0, 0, 0.1)` in the [color panel](https://github.com/WordPress/gutenberg/blob/8675f86f885ca389f3bca39971b22cb897bab61b/packages/block-editor/src/hooks/color.scss#L34-L42)

3. **Review the style overrides in `ComplementaryArea`.**  Despite the changes from the points above, the separator in the Global Styles sidebar will keep using `$gray-100` because of the [style overrides from `ComplementaryArea`](https://github.com/WordPress/gutenberg/blob/c1ea8d60db9f05ebbfbaced50a88bfe30a7a0270/packages/interface/src/components/complementary-area/style.scss#L55). We could start by changing the color used by these style overrides, or we could work on a more structured solution (see https://github.com/WordPress/gutenberg/pull/40526#discussion_r855973432).

Comment (ciampo) [reply packages/edit-site/src/components/sidebar/style.scss]: I will remove this line (as explained in https://github.com/WordPress/gutenberg/pull/40526#discussion_r856006044).

We can continue the conversation there, regarding which shade of gray should be used going forward

**Comment by ciampo** [comment]
I removed the border color override, this is how the sidebar looks after this change:

Screenshot 2022-04-22 at 12 14 14


I will wait for CI to go green and merge.

File: packages/edit-site/src/components/sidebar/style.scss (hunk 2 above)
Comment (jasmussen) [reply packages/edit-site/src/components/sidebar/style.scss]: Ah, excellent context. It does sound like we'll want to fix that at the component level.

At the _moment_ we have these shades of gray for most of the UI:

```
$gray-900: #1e1e1e;
$gray-800: #2f2f2f;
$gray-700: #757575;		// Meets 4.6:1 text contrast against white.
$gray-600: #949494;		// Meets 3:1 UI or large text contrast against white.
$gray-400: #ccc;
$gray-300: #ddd;		// Used for most borders.
$gray-200: #e0e0e0;		// Used sparingly for light borders.
$gray-100: #f0f0f0;		// Used for light gray backgrounds.
```

In this case, the 0.1 opacity divider actually matches `$gray-100`, but it ideally _should_ match `$gray-200` as those are used on panels and general UI.

I understand there are challenges with making the components work on a dark background, and that it can therefore make sense to use more agnostic colors that aren't optimized for a white background, perhaps even opacities. I trust that'll be taken into account 👍 👍
